.github/workflows/ci.yml

```
@@ -0,0 +1,52 @@
+      if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+      run: poetry install --no-interaction --no-root --extras "all"
+
+    - name: Install project
+      run: poetry install --no-interaction --extras "all"
+
+    - name: Run tests
+      run: poetry run pytest tests/ -v
+
+    - name: Run tests with coverage
+      run: |
+        poetry run pytest tests/ --cov=pyspark_datasources --cov-report=xml --cov-report=term-missing
```

YAMLlint (1.37.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)